search/tests/mock_search_engine.py

```diff
@@ -149,54 +149,60 @@ def _process_exclude_dictionary(documents_to_search, exclude_dictionary):
     return documents_to_search
 
 
-def _count_facet_values(documents, facet_terms):
+def _count_aggregated_values(documents, agg_terms):
     """
-    Calculate the counts for the facets provided:
+    Calculate the counts for the aggregations provided:
 
-    For each facet, count up the number of hits for each facet value, so
+    For each aggregation, count up the number of hits for each aggregation value, so
     that we can report back the breakdown of how many of each value there
-    exist. Notice that the total is the total number of facet matches that
+    exist. Notice that the total is the total number of aggregation matches that
     we receive - a single document will get counted multiple times in the
-    total if the facet field is multi-valued:
+    total if the aggregation field is multi-valued:
 
         e.g. a course may have a value for modes as ["honor", "validated"], and
         so the document will count as 1 towards the honor count, 1 towards the
         validated count, and 2 towards the total. (This may be a little
         surprising but matches the behaviour that elasticsearch presents)
 
     """
-    facets = {}
+    aggregations = {}
 
-    def process_facet(facet):
-        """ Find the values for this facet """
-        faceted_documents = [facet_document for facet_document in documents if facet in facet_document]
+    def process_aggregation(aggregate):
+        """
+        Find the values for this aggregation
+        """
+        aggregated_documents = [
+            agg_document for agg_document in documents if aggregate in agg_document
+        ]
         terms = {}
 
-        def add_facet_value(facet_value):
-            """ adds the discovered value to the counts for the selected facet """
-            if isinstance(facet_value, list):
-                for individual_value in facet_value:
-                    add_facet_value(individual_value)
+        def add_agg_value(agg_value):
+            """
+            adds the discovered value to the counts for the selected aggregation
+            """
+            if isinstance(agg_value, list):
+                for individual_value in agg_value:
+                    add_agg_value(individual_value)
             else:
-                if facet_value not in terms:
-                    terms[facet_value] = 0
-                terms[facet_value] += 1
+                if agg_value not in terms:
+                    terms[agg_value] = 0
```

Comment:
oh, I see the pain. Please use `defaultdict(int)` and just breathe as an honored man.

Reply:
Done